### ISSUE TITLE
Issue Running Server

### ISSUE THREAD

**Issue by krazzarkov** [branden-akana/shen#21]
Hello,

I saw this project and liked the idea of it so I tried running it to see how it looks/performs.

Unfortunately, even after creating my .env file I cannot start the server without getting an error.

This is a pastebin of the log generated when I try and run the server:[https://pastebin.com/nG1BPtzz](https://pastebin.com/nG1BPtzz)

Additionally, this is everything in my .env file.

FIREBASE_DATABASE_URL = "firebaseurl"
FIREBASE_PROJECT_ID = "firebaseprojectid"
FIREBASE_CLIENT_EMAIL = "myemail"
FIREBASE_PRIVATE_KEY = "myprivatekey"

DISCORD_CLIENT_ID = "discodclientid"
DISCORD_CLIENT_SECRET = "clientsecret"
DISCORD_BOT_TOKEN = "bottoken"

Any help would be appreciated.

Regards,
Kraz

**Comment by bakana808** [comment]
Sorry about that!

Awhile ago I've had the project run off a PostgreSQL database instead of Firebase, but I guess I seem to have forgotten to update the README. I do plan to support both of them as a method of storing data if Firebase is what you prefer, but the PostgreSQL implementation is what's being worked on at the moment.

Still, the server not being able to start without any kind of clear error message is pretty important, so I'll add in some error messages soon.

If you're curious, the environment variable containing your PostgreSQL database information is `DATABASE_URL`. There are some other variables not mentioned in the README, but some commits are coming soon that should just give you warnings if they're missing.

Anyways thanks for your interest and yeah, unfortunately this project isn't quite ready yet!

**Comment by krazzarkov** [comment]
Hello,

Thanks for the quick reply. 

I am not sure what I would want to use for this as I haven't really seen how the site performs and functions when it is started up yet. I am however more familiar with Firebase so I would probably prefer to use that for the time being.

I also just wanted to confirm, the current version of the project does not support Firebase?

**Comment by bakana808** [comment]
No, not right now. Sorry.

**Comment by krazzarkov** [comment]
That's ok.

If I want to test this with PostgreSQL and I setup a local database do I just need to insert the DATABASE_URL into the .env? Asking since you mentioned that there are more variables.

**Comment by bakana808** [comment]
Yes, the DATABASE_URL is for PostgreSQL and should contain the username and
password as well as the URL.

The other variables are the id/secret for a Google app (for authentication)
and the server ID to use for the Discord bot (which i should just take out
later), but the server should run just fine without those variables. It'll
just give you some warnings.

**Comment by krazzarkov** [comment]
Sounds good,

I'll try and test it later today, Thank you for the help!

**Comment by bakana808** [comment]
Anytime.

By the way, can you confirm if the server doesn't crash anymore so I can
call this issue closed? It should run even without a database (but some of
the commands won't work).

**Comment by krazzarkov** [comment]
Well the thing is that it crashes no matter what. 

I have tried removing the .env file and I have tried removing the part about firebase but it keeps giving me the exact same error as I originally had.

Haven't tested to see if it will error out with the PostgreSQL database_url yet though. However, if you still want to close the issue you're welcome to do so.
